### ISSUE TITLE
Tolerance in StandardFlexibleScaler should be relative

### ISSUE THREAD

**Issue by Luthaf** [cosmo-epfl/scikit-cosmo#82]
I have these properties I want to scale before using KPCovR:
```
Y = [
 [5.500e+00 2.400e-29]
 [3.600e+00 2.270e-28]
 [3.750e+00 2.640e-28]
 [3.020e+00 6.700e-29]
 [9.175e-01 8.300e-32]
]
```

One column only contains very small values, leading to the following exceptions, which is a bit surprising.
```
Traceback (most recent call last):
  File "do-kpocvr.py", line 119, in <module>
    Y = StandardFlexibleScaler(column_wise=True).fit_transform(Y)
  File "/opt/miniconda3/lib/python3.7/site-packages/skcosmo/preprocessing/flexible_scaler.py", line 86, in fit_transform
    self.fit(X, y)
  File "/opt/miniconda3/lib/python3.7/site-packages/skcosmo/preprocessing/flexible_scaler.py", line 50, in fit
    raise ValueError("Cannot normalize a feature with zero variance")
ValueError: Cannot normalize a feature with zero variance
```

I'll change the `tol` parameter to make this work for now, but I think that instead of checking the absolute value of the variance against the tolerance, we should test the relative variance, i.e. replace `np.any(var < self.tol)` with `np.any(var / np.mean(Y, axis=0) < self.tol)` in https://github.com/cosmo-epfl/scikit-cosmo/blob/64969332914aa956e943c8f1824699731510d72d/skcosmo/preprocessing/flexible_scaler.py#L60-L62

**Comment by bhelfrecht** [comment]
We could do something similar to what `numpy` often does, and use both an absolute and a relative tolerance, e.g., https://numpy.org/doc/stable/reference/generated/numpy.allclose.html

**Comment by hurricane642** [comment]
This problem was solved in #91
